stomp_core/src/stomp.cpp

```diff
@@ -590,26 +592,25 @@ bool Stomp::computeNoisyRolloutsCosts()
 
 bool Stomp::computeRolloutsStateCosts()
 {
-
-  bool all_valid = true;
   bool proceed = true;
+
+  #pragma omp parallel for shared(proceed)
   for(auto r = 0u ; r < config_.num_rollouts; r++)
   {
-    if(!proceed_)
+    if(proceed)
```

Comment:
Yea, I could not find a way around this because you cannot share a member variable between between multiple threads. @Jmeyer1292 is there a way around this; could a global static variable be used?

Reply:
It looks like we should make it a global variable which then can be accessed.

Reply:
Would changing a global variable cause unintended interruptions of other running instances of STOMP? If so we may try to setup a  static map variable with and int **id** as the key and a boolean value.  So each instance of stomp gets assigned a unique ID  that is then used in the ompl for-loop to lookup the boolean corresponding to that instance.

Reply:
I think we would still have issues when making it a global in a multi threaded use case of the library. I think the current implementation is the best that can be achieved using OpenMP.  It will still cancel if one of the threads experience an error just not when requested externally. This may not be an issue given how long it takes to run this method though.

Reply:
That may be true for environments with few complex geometries but in more complex environments   or specific stomp configurations (iterations, rollouts, etc.) could it still take a while to produce a plan?